packages/zoe/contracts/helpers/bondingCurves.js

```diff
@@ -0,0 +1,77 @@
+import { natSafeMath } from './safeMath';
+
+const { add, subtract, multiply, divide } = natSafeMath;
+
+/**
+ * Contains the logic for calculating how many units should be given
+ * back to the user in exchange for what they sent in. It also
+ * calculates the new units of the assets in the pool. Reused in
+ * several different places, including to check whether an offer is
+ * valid, getting the current price for an asset on user request, and
+ * to do the actual reallocation after an offer has been made.
+ * @param  {object} zoe - the contract facet of Zoe
+ * @param  {assay[]} assays - an array of assays
+ *
+ * Once the function is made, it has the following parameters:
+ * @param  {units[]} poolUnitsArray - an array of the current units in the
+ * liquidity pool
+ * @param  {units} unitsIn - the units sent in by a user
+ * @param  {number} feeInTenthOfPercent=3 - the fee taken in tenths of
+ * a percent. The default is 0.3%. The fee is taken from unitsIn
+ */
+
+export const makeCalculateConstProductFn = (zoe, assays) => {
+  const unitOpsArray = zoe.getUnitOpsForAssays(assays);
+
+  return (poolUnitsArray, unitsIn, feeInTenthOfPercent = 3) => {
```

Comment:
Should validate that the `feeInTenthOfPercent` is a Nat < 1000. I didn't get that you were expecting a Nat. Both from the API and the comment I thought you were expecting a float.

OTOH, should it be a float, not scaled, that must be between 0 and 1? Then, to represent 0.3%, you'd just say 0.003. I don't understand how this would change roundoff considerations, but my guess is that it doesn't makes things any worse than they already must be, and it would let you drop all the artificial 1000x scaling.

Reply:
The uniswap code hard-codes 997 (1000 - 3). Although we shouldn't also hard-code it, I like keeping the idea that it's a Nat rather than a float. I'd like to keep to their code as much as possible.